### PR TITLE
Switch filter tags input to a button

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Switch filter tags input to a button

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Clicky the button and verify all works as expected

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1081

### CODE REVIEW

**Comment by hatched** [comment]
https://jaas-dashboard-canonical-web-and-design-pr-363.run.demo.haus

**Comment by hatched** [comment]
QA Notes:
- It would be nice if the chevron flipped when the drop down menu was open.
- If the drop down is open, clicking the button again doesn't close it.

File: src/components/FilterTags/FilterTags.js [hunk 1]
```diff
@@ -154,23 +153,23 @@ const FilterTags = () => {
...
+        <span className="p-filter-tags__active-count">
+          {activeFilters.length
             ? `Active filters: ${activeFilters.length}`
-            : "Filter models"
-        }
-      />
-
+            : "Filter models"}
+        </span>
+        <i className="p-icon--contextual-menu"></i>
```
Comment (anthonydillon) [inline src/components/FilterTags/FilterTags.js]: Can we not just use the vanilla component for this? https://docs.vanillaframework.io/patterns/contextual-menu#indicator

**Comment by barrymcgee** [comment]
@anthonydillon The short answer from my end is that I wasn't aware of that pattern. I spoke with @spencerbygraves about this in stand-up and he has concerns that the contextual menu is green by default and the width is significantly narrower than what we have now. I asked that he speak to you directly about this and add the preferred course of action here.

**Comment by anthonydillon** [comment]
It's green only because it's using the positive button. Change that to what you want. It's it worth building something brand new to be a few pixels wider? If so, you use the grid to set the width of the parent and use something like `.row` or create a state of `is-wide` to set the width to 100%.

**Comment by spencerbygraves** [comment]
@barrymcgee @anthonydillon @clagom we could use contextual menu with neutral button styling and the normal icon colour.

We don't have a dense version so it will be bigger than what we have now. 

image

**Comment by anthonydillon** [comment]
We can create native Vanilla dense contextual menus:
https://codepen.io/anthonydillon/pen/XWbRWvy

**Comment by clagom** [comment]
> We can create native Vanilla dense contextual menus:
> https://codepen.io/anthonydillon/pen/XWbRWvy
Sounds good

**Comment by barrymcgee** [comment]
Screenshot 2020-02-27 at 14 05 17
Screenshot 2020-02-27 at 14 05 48

**Comment by anthonydillon** [comment]
## QA issue
- Load models
- Open the filter windows and select owner: gomboli
- **All models are displayed** _Should only display gomboli models_
- Refresh the page
- **All models are displayed** _Should only display gomboli models_
- Filter input says: **Active filters: 14** _Should say filtered by 1 thing_
- Open the filter and add another filter of owner: ya-bo-ng
- **The original filter term is exploded by character** _Should append owner: ya-bo-ng to the filters_


**ERRORS** _EXPECTED_

**Comment by barrymcgee** [comment]
@anthonydillon It appears this the demo link associated with this PR is running an outdated version of the site. None of the bugs you list above replicate when I run this branch locally... can you confirm by running locally yourself?

File: src/components/FilterTags/FilterTags.js [hunk 2]
```diff
@@ -153,82 +152,108 @@ const FilterTags = () => {
...
-            filters[filterBy].length > 0 && (
-              <div key={filterBy} className="p-filter-panel__section">
-                <h4 className="p-filter-panel__heading">
-                  {pluralize(filters[filterBy].length, filterBy)}
-                </h4>
-                <ul
-                  className="p-list p-filter-panel__list"
-                  data-test={filterBy}
+                  {activeFilter.replace(":", ": ")}
+                  <i
```
Comment (anthonydillon) [inline src/components/FilterTags/FilterTags.js]: Can you make this a link element? It seems to be the only thing that doesnt make via keyboard.

Comment (barrymcgee) [reply src/components/FilterTags/FilterTags.js]: Done!

**Pull request merged**